src/mixins/date-mixin.ts

```diff
@@ -1,81 +1,174 @@
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins } from 'vue-property-decorator'
+import { Getter } from 'vuex-class'
+import { isDate } from 'lodash'
+import { CommonMixin } from '@/mixins'
 
 /**
  * Mixin that provides some useful date utilities.
  */
 @Component({})
-export default class DateMixin extends Vue {
+export default class DateMixin extends Mixins(CommonMixin) {
+  @Getter getCurrentJsDate!: Date
+  @Getter getCurrentDate!: string
   readonly MS_IN_A_DAY = (1000 * 60 * 60 * 24)
 
   /**
-   * Converts a JavaScript date object to a simple date string.
-   * @param date The date to convert.
-   * @returns A simple date string formatted as YYYY-MM-DD.
+   * Fetches and returns the web server's current date (in UTC).
+   * Used to bypass the user's local clock/timezone.
+   * Ref: https://www.npmjs.com/package/serverdate
+   * @returns a promise to return a Date object
    */
-  dateToUsableString (date: Date): string {
-    if (!date || date.toString() === 'Invalid Date') return null
+  async getServerDate (): Promise<Date> {
+    const input = window.location.origin
+    const init: RequestInit = { cache: 'no-store', method: 'HEAD' }
+
+    // don't call fetch() during Jest tests
+    // because it's not defined
+    if (this.isJestRunning) return new Date()
+
+    const { headers, ok, statusText } = await fetch(input, init)
+
+    if (!ok) {
+      // eslint-disable-next-line no-console
+      console.warn('Unable to get server date - using browser date instead')
+      // fall back to local date
+      // NB: filing  may contain invalid dates/times
+      return new Date()
+    }
+
+    return new Date(headers.get('Date'))
+  }
+
+  /**
+   * Creates and returns a new Date object in UTC, given parameters in Pacific timezone.
+   * (This works regardless of user's local clock/timezone.)
```

Comment:
Otherwise, JS creates a Date (which is [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date](internally stored in milliseconds since Jan 1, 1970 UTC))... but how does JS know what UTC is? It uses the local timezone offset, of course... which could be spoofed or wrong.

Therefore, this method starts with a known UTC date (from server) and then creates a Pacific date object from it.